test/e2e/upgrade_test.go

```diff
@@ -37,6 +37,15 @@ func runSimpleUpgradeFlow(test *framework.ClusterE2ETest, updateVersion v1alpha1
 	test.DeleteCluster()
 }
 
+func runUpgradeFlowWithAPI(test *framework.ClusterE2ETest, clusterOpts ...framework.ClusterE2ETestOpt) {
+	test.GenerateClusterConfig()
+	test.CreateCluster()
+	test.UpgradeClusterWithKubectl(clusterOpts)
+	test.ValidateClusterState()
```

Comment:
I think this should be inside `UpgradeClusterWithKubectl` command, since we'll always need to call it. It's basically apart of the process.